base/src/main/scala/data/ProCompose.scala

```diff
@@ -0,0 +1,54 @@
+package scalaz
+package data 
+
+object ProCompose {
+  import typeclass.{Profunctor, Strong, Category, StrongClass, CategoryClass}
+  import typeclass.Strong._
+  import typeclass.Profunctor._
+
+  trait ProCompose[P[_, _], A, B] { } 
+
+  case class Pure[P[_, _], A, B](pab: A => B) extends ProCompose[P, A, B]
+
+  case class Composed[P[_, _], A, X, B](pab: P[A, X], more: ProCompose[P, X, B]) extends ProCompose[P, A, B]
```

Comment:
> Try to implement it yourself

What is "it"? What is given to me and what should I try to implement?

Seems like you have one specific usage in mind that is not stack-safe, and even that only when "you do it the obvious way". That is different than saying "there is no stack-safe usage of this structure".

For example, this structure supports balanced composition (e.g. of `Kleisli`s), which will be stack safe for all practical instances of `ProCompose[Kleisli[F, ?, ?], A, B]`.

Also, using this with a something like `Evaluable` seems like a natural use of `ProCompose`. Do you have a usage in mind that doesn't necessarily involve `Evaluable`, but cannot be made stack-safe without it?